### PR TITLE
Do not allow an alias to override an existing package

### LINKED ISSUE #4423: Prevent conan alias from overwriting existing packages
When creating an alias package with `conan alias` it is easy to mix up the order of parameters:

* Correct: `conan alias boost/ALIAS@conan/stable boost/1.69.0@conan/stable`
* Wrong: `conan alias boost/1.69.0@conan/stable boost/ALIAS@conan/stable`

In case you're accidentally calling the wrong order you're overwriting the package you initially wanted to reference. If this happens you render your original package useless and need to remove it. Then you can start all over again.
It would be perfect to have a built-in check that tells the user that the existing package `boost/1.69.0@conan/stable` already exists and will be overwritten by an alias package - and if the user is really sure this is what he wants to do.

Happens to me with Conan 1.11.0.

### PR DESCRIPTION
**Pull request by jgsogo**
Changelog: Fix: Do not allow an alias to override an existing package
Docs: omit

closes #4423 

✅ Do not pass the `cache` object to the `export_alias` function.

@REVISIONS: 1

### CODE REVIEW

File: conans/client/cmd/export.py [hunk 1]
```diff
@@ -15,29 +15,25 @@
...
-    if reference.name != target_reference.name:
-        raise ConanException("An alias can only be defined to a package with the same name")
+def export_alias(ref_layout, target_reference, output, revisions_enabled):
     conanfile = """
 from conans import ConanFile
 
 class AliasConanfile(ConanFile):
     alias = "%s"
-""" % target_reference.full_repr()
+""" % target_reference
```
Comment (memsharded) [inline conans/client/cmd/export.py]: This is changing behavior, not allowing alias to revisions, why? Discuss with @lasote what behavior we want for alias.

File: conans/client/cmd/export.py [hunk 2]
```diff
@@ -15,29 +15,25 @@
...
-    mkdir(cache.export_sources(reference))
-    digest = FileTreeManifest.create(export_path)
-    digest.save(export_path)
+    save(ref_layout.conanfile(), conanfile)
+    digest = FileTreeManifest.create(ref_layout.export())
+    digest.save(folder=ref_layout.export())
 
     # Create the metadata for the alias
-    _update_revision_in_metadata(cache, output, None, reference, digest)
+    _update_revision_in_metadata(ref_layout=ref_layout, revisions_enabled=revisions_enabled,
```
Comment (memsharded) [inline conans/client/cmd/export.py]: Changing this to use layouts should have been done in the other PR, not in this one.

File: conans/client/cmd/export.py (hunk 1 above)
Comment (lasote) [reply conans/client/cmd/export.py]: I relaunched the CI for this with revisions enabled, because yes, it should break the alias test with revisions.

**Comment by jgsogo** [comment]
That's it, time to review and merge =D

**Pull request merged**